pkg/operator/image.go

```diff
@@ -56,6 +56,32 @@ func BuildImagePath(specImage, baseImage, version, tag, sha string) (string, err
 	return baseImage + ":" + version, nil
 }
 
+// BuildImagePathForAgent builds a container image path based on
+// the given parameters for Prometheus Agent.
+// Return specImage if not empty.
+// If image contains a tag or digest then image will be returned.
+// Otherwise, return image suffixed by either SHA, tag or version(in that order).
+// Inspired by kubernetes code handling of image building.
+func BuildImagePathForAgent(specImage, baseImage, version string) (string, error) {
```

Comment:
it could also call `BuildImagePath()` directly with empty strings for arguments which are irrelevant?

Reply:
ah ok, now I fully understand what you mean in https://github.com/prometheus-operator/prometheus-operator/pull/6686#discussion_r1645778377